draft-liu-teas-transport-network-slice-yang.txt

```diff
@@ -0,0 +1,1568 @@
+
+
+   Internet-Drafts are draft documents valid for a maximum of six months
+   and may be updated, replaced, or obsoleted by other documents at any
+   time.  It is inappropriate to use Internet-Drafts as reference
+   material or to cite them other than as "work in progress."
+
+   This Internet-Draft will expire on 13 August 2023.
+
+Copyright Notice
+
+   Copyright (c) 2023 IETF Trust and the persons identified as the
+   document authors.  All rights reserved.
+
+   This document is subject to BCP 78 and the IETF Trust's Legal
+   Provisions Relating to IETF Documents (https://trustee.ietf.org/
+   license-info) in effect on the date of publication of this document.
+   Please review these documents carefully, as they describe your rights
+   and restrictions with respect to this document.  Code Components
+   extracted from this document must include Revised BSD License text as
+   described in Section 4.e of the Trust Legal Provisions and are
+   provided without warranty as described in the Revised BSD License.
+
+Table of Contents
+
+   1.  Introduction  . . . . . . . . . . . . . . . . . . . . . . . .   3
+     1.1.  Terminology . . . . . . . . . . . . . . . . . . . . . . .   3
+     1.2.  Tree Diagram  . . . . . . . . . . . . . . . . . . . . . .   4
+     1.3.  Prefixes in Data Node Names . . . . . . . . . . . . . . .   4
+   2.  Modeling Considerations . . . . . . . . . . . . . . . . . . .   4
+     2.1.  Relationships to Related Topology Models  . . . . . . . .   4
+     2.2.  ACTN for Network Slicing  . . . . . . . . . . . . . . . .   5
+   3.  Model Applicability . . . . . . . . . . . . . . . . . . . . .   5
+   4.  YANG Model Overview . . . . . . . . . . . . . . . . . . . . .   8
+   5.  Model Tree Structure  . . . . . . . . . . . . . . . . . . . .   8
+   6.  YANG Modules  . . . . . . . . . . . . . . . . . . . . . . . .  11
+   7.  Manageability Considerations  . . . . . . . . . . . . . . . .  18
+   8.  Security Considerations . . . . . . . . . . . . . . . . . . .  19
+   9.  IANA Considerations . . . . . . . . . . . . . . . . . . . . .  19
+   10. References  . . . . . . . . . . . . . . . . . . . . . . . . .  20
+     10.1.  Normative References . . . . . . . . . . . . . . . . . .  20
+     10.2.  Informative References . . . . . . . . . . . . . . . . .  21
+   Appendix A.  Acknowledgments  . . . . . . . . . . . . . . . . . .  22
+   Appendix B.  Data Tree for the Example in Section 3.1 . . . . . .  22
+     B.1.  Native Topology . . . . . . . . . . . . . . . . . . . . .  23
+   Authors' Addresses  . . . . . . . . . . . . . . . . . . . . . . .  27
+
+
+
+
+
+
+
+Liu, et al.              Expires 13 August 2023                 [Page 2]
+
+Internet-Draft      Network Slice Topology Data Model      February 2023
+
+
+1.  Introduction
+
+   This document defines a YANG [RFC7950] data model for representing,
+   managing, and controlling IETF network slices as abstract network
+   topologies, where the network slices are defined in
```

Comment:
"YANG data model for representing ... IETF network slices as abstract network topologies"-> how far can be such representation different from NRP? would it not be the NRP the kind of topology considered for slicing?
I'm asking this because whatever topology should be consistent (in my view) with the NRP supporting it. Then there could be dependencies between this slice topology data model and the NRP data model.What do you think?

Reply:
An NRP is a realization of a network slice topology within the NSC.
A NS topology is customer intent, whereas NRP is a realization, topology != NRP
Use of NRP could be optional
topology describes how client traffic runs and NRP describes how resources are grouped to support one or multiple network slices. NRP may or may not be a topology in its own right